src/utils/supportedTextChannel.ts

```diff
@@ -0,0 +1,10 @@
+import { ChannelType } from 'discord.js'
+
+// Define a custom type called `supportedTextChannel` that represents the supported channel types
+export type supportedTextChannel =
+  | ChannelType.GuildText
+  | ChannelType.GuildVoice
+  | ChannelType.GuildStageVoice
+  | ChannelType.PublicThread
+  | ChannelType.PrivateThread
+  | ChannelType.AnnouncementThread
```

Comment:
Discord.js has `GuildTextBasedChannelTypes` defined as:

```ts
type GuildTextBasedChannelTypes =
  | ChannelType.GuildText
  | ChannelType.GuildVoice
  | ChannelType.GuildAnnouncement
  | ChannelType.AnnouncementThread
  | ChannelType.PublicThread
  | ChannelType.PrivateThread
  | ChannelType.GuildStageVoice
```

not sure if we should use that type instead of enumerating it here

Reply:
oh we very should use that krubb